### PR TITLE
better commonjs test (clean dist, consistent test)

### LINKED ISSUE #1909: running tests from fresh repos commonly fails
Reproduce:
directly after pulling and running `npm install && bower install && grunt test` the test results with 

Expected Results:
Assuming last merge was good all tests should pass.

Results:
```
Testing http://localhost:9000/test/commonjs.html
>> PhantomJS timed out, possibly due to:
```
commonjs unit tests fail because of missing `test/commonjs-bundle.js` resource are not available when running `grunt test`

### PR DESCRIPTION
**Pull request by swilliamset**
fix #1909
- removed commonjs testing resource (js/npm.js) from dist folder
- buld dist task no longer leaves dist/js/npm.js in zip
- moved creation of resources for commonjs into prep test task
- test dist task now preps for commonjs tasks and cleans up

### CODE REVIEW

File: grunt/tasks/test.js [hunk 1]
```diff
@@ -1,15 +1,21 @@
 module.exports = function test (grunt) {
 	// to be run prior to submitting a PR
 	grunt.registerTask('test', 'run jshint, qunit source w/ coverage, and validate HTML',
-		['browserify:commonjs', 'dist', 'jshint', 'connect:testServer', 'qunit:noMoment', 'qunit:globals', 'qunit:dist', 'htmllint', 'resetdist']);
+		['jshint', 'connect:testServer', 'qunit:noMoment', 'qunit:globals', 'test-dist', 'htmllint']);
+
+	grunt.registerTask('prep-commonjs-test', 'run commonjs config buiild and browserify to prep for commonjs test',
```
Comment (futuremint) [inline grunt/tasks/test.js]: Typo in description `buiild`.

File: grunt/tasks/build.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ module.exports = function(grunt) {
 
 	// Full distribution task
 	grunt.registerTask('dist', 'Build "dist." Contributors: do not commit "dist."',
-			['clean:dist', 'distcss', 'copy:fonts', 'copy:templates', 'distjs', 'commonjs', 'distzip']);
+			['clean:dist', 'distcss', 'copy:fonts', 'copy:templates', 'distjs', 'distzip']);
```
Comment (futuremint) [inline grunt/tasks/build.js]: Is the commonjs bundle used anywhere else in the dev stuff?  Meaning... is it _ok_ that we're removing it from dist completely and only running it during test?

Comment (swilliamset) [inline grunt/tasks/build.js]: no it's not used elsewhere

**Comment by vernak2539** [comment]
just wondering, if you're deleting it, will it be re-added during publishing?

https://github.com/ExactTarget/fuelux/blob/master/package.json#L74

File: grunt/tasks/build.js (hunk 2 above)
Comment (interactivellama) [inline grunt/tasks/build.js]: It's used by CommonJS users of Fuel UX and anyone that pulls in FuelUX via NPM I would assume.

**Pull request merged**